### ISSUE TITLE
Preview dropdown menu missing accessible name

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#41514]
### Description

In both the post and the site editors, clicking the "Preview" button opens up a dropdown menu, which uses the "menu" ARIA role. This menu should have an accessible name, using either `aria-label` or `aria-labelledby`, but currently it doesn't have one. The accessible name helps screen readers identify the purpose of the menu.

To fix this, a `label` prop can be passed to the DropdownMenu component used in `packages/block-editor/src/components/preview-options/index.js`.

### Step-by-step reproduction instructions

1. Go to the post or the site editor and click "Preview" button in top toolbar.
2. Open up browser code inspector.
3. Check that preview dropdown markup doesn't have an accessible label.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
